### PR TITLE
feature: first attempt at task table

### PR DESCRIPTION
**Pull request by tguar**
Looking to create a task table that can support the following:

1. a task has an id
2. a task has a title
3. a task has weight
4. a task can repeat every x days 
5. a task can repeat between floor <= x <= ceiling
6. a task can be deferred

### CODE REVIEW

**Comment by tguar** [comment]
I think UserId should be a foreign key, right now it's commented out because I didn't know how to properly link it

File: src/models/tasks.model.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  // })
+  // userId: string;
+
+  @AllowNull(false)
+  @Column(DataType.STRING(40))
+  title: string;
+
+  @AllowNull(false)
+  @Column(DataType.DECIMAL(1, 2))
+  UserWeight: number;
```
Comment (joewinnn) [inline src/models/tasks.model.ts]: why are some fields uppercase but others lowercase? ex) taskId, title, and UserWeight?

**Comment by joewinnn** [comment]
Maybe I still don't understand how you're implementing the "repeating every x days"

I see fields for ceiling and floor, but where's the "x" field?

File: src/models/tasks.model.ts (hunk 1 above)
Comment (tguar) [reply src/models/tasks.model.ts]: Because I didn't proofread this before I pushed haha. Let me fix that

**Comment by tguar** [comment]
> Maybe I still don't understand how you're implementing the "repeating every x days"
> 
> I see fields for ceiling and floor, but where's the "x" field?

I don't think x is defined by a value. x is just a task that falls between the two boundaries if its a range, else if its repeat every x days the ceiling is null and the floor is the value it repeats on

File: src/models/tasks.model.ts (hunk 1 above)
Comment (tguar) [reply src/models/tasks.model.ts]: Changed, I'll let you resolve

Comment (joewinnn) [reply src/models/tasks.model.ts]: Not given an option to resolve.  Maybe @jin0s add me as collaborator?

Comment (jin0s) [reply src/models/tasks.model.ts]: my bad, I added your email instead of your github account

**Comment by jin0s** [comment]
> > Maybe I still don't understand how you're implementing the "repeating every x days"
> > I see fields for ceiling and floor, but where's the "x" field?
> 
> I don't think x is defined by a value. x is just a task that falls between the two boundaries if its a range, else if its repeat every x days the ceiling is null and the floor is the value it repeats on

@joewinnn Tom and I discussed it briefly, if you want "repeating every x days" this can be achieve by setting the repeating floor and ceiling numbers to the same value x. That way we get both benefits of repeating it on a specific number of days and also repeating within a range of days.

**Pull request merged**